pkg/sql/logictest/testdata/logic_test/fk

```diff
@@ -2715,9 +2715,16 @@ statement ok
 CREATE TABLE child_duplicate_cols (id INT, parent_id INT, PRIMARY KEY (id))
 
 # The fk constraint is invalid because it has duplicate columns, so automatically adding the index fails
+# Unfortunately, the error message is slightly different between legacy and declarative schema changer
+# so we have the same statement twice here.
+onlyif config local-legacy-schema-changer
 statement error foreign key contains duplicate column \"parent_id\"
 ALTER TABLE child_duplicate_cols ADD CONSTRAINT fk FOREIGN KEY (parent_id, parent_id) references parent
 
+skipif config local-legacy-schema-changer
+statement error pq: foreign key contains duplicate column "test.public.child_duplicate_cols.parent_id"
```

Comment:
Can you test for the correct pgcode here? Postgres uses `42830`.

Reply:
done